src/Microsoft.Health.Dicom.Core/Features/Update/UpdateInstanceOperationService.cs

```diff
@@ -95,11 +103,11 @@ public class UpdateInstanceOperationService : IUpdateInstanceOperationService
         EnsureArg.IsNotNull(updateSpecification, nameof(updateSpecification));
         EnsureArg.IsNotNull(updateSpecification.ChangeDataset, nameof(updateSpecification.ChangeDataset));
 
-        int partitionKey = _contextAccessor.RequestContext.GetPartitionKey();
+        Partition partition = _contextAccessor.RequestContext.GetPartition();
 
         try
         {
-            var operation = await _client.StartUpdateOperationAsync(operationId, updateSpecification, partitionKey, cancellationToken);
+            var operation = await _client.StartUpdateOperationAsync(operationId, updateSpecification, partition, _externalStoreEnabled, cancellationToken);
```

Comment:
We should not be passing around externalStoreEnabled variable. The setting is static in context of a dicom Service. 
The config can be read in Functions process and ideally during registration of a service, things happen correctly.

Passing around properties makes sense only when things are dynamic per request.